### PR TITLE
Monthly habits

### PR DESCRIPTION
**Pull request by WMudgeEllis**
# Change Log

- Closes # 45

## What I did:

- Exposed monthly habit entries

## Things to consider:

- Same format as weeklyHabit query, not sure if that is desired

## Test coverage

- [x] 100% Model Testing
- [x] 100% Request Testing

If below 100%, what still needs testing:

-

### CODE REVIEW

File: app/graphql/types/user_type.rb [hunk 1]
```diff
@@ -14,6 +14,9 @@ class UserType < Types::BaseObject
     field :monthly_moods, [Types::MoodType], null: true do
       argument :month, Int, required: false
     end
+    field :monthly_habits, [Types::HabitEntryType], null: true do
```
Comment (WMudgeEllis) [inline app/graphql/types/user_type.rb:17]: I'm not sure what the null: true does. Is it needed for this query?

**Review by TannerDale — APPROVED** [review]
Nice looks good

File: app/models/user.rb [hunk 2]
```diff
@@ -21,6 +21,12 @@ def weekly_habits
       .order('habit_entries.created_at DESC')
   end
 
+  def monthly_habits(month)
+    habit_entries
+      .where('extract(month from habit_entries.created_at) = ? AND habit_entries.status = 1', month)
+      .order('habit_entries.created_at DESC')
+  end
```
Comment (WMudgeEllis) [inline app/models/user.rb:28]: This query does not have the group clause that the other habit entry one does, in fact the other one doesn't need it either (at least all tests passed without it).

File: app/graphql/types/user_type.rb (hunk 1 above)
Comment (TannerDale) [reply app/graphql/types/user_type.rb:17]: I allows the field to return nil for a query I'm pretty sure. I'm not sure if we need that specifically for this query. If there are no habits completed for that month it should return an empty array

**Review by Lmcourt — APPROVED** [review]

**Pull request merged**